### PR TITLE
Added optional Host to DaprConnectionString

### LINKED ISSUE #2235: Enable the definition of the GRPS endpoint host
**Is your feature request related to a problem? Please describe.**
Since our DAPR sidecar is not accessible via "localhost" we need to define the host of the GrpcEndpoint. Currently, its hardcoded "localhost".

**Describe the solution you'd like**
Add "Host" as additional property within the `DaprConnectionString` like "DaprConnectionString=GrpcPort=6011;Host=10.1.1.1" and use it to build the `GrpcEndpoint` and `HttpEndpoint`.

### PR DESCRIPTION
**Pull request by Haeavar**
This PR adds an optional Host to DaprConnectionString like "DaprConnectionString=PubSubComponent=pubsub;GrpcPort=6011;Host=10.1.1.1".

Will fix #2235

### CODE REVIEW

**Review by marcschier — APPROVED** [review]

**Comment by marcschier** [comment]
Many thanks for the contribution @Haeavar

File: docs/opc-publisher/commandline.md [hunk 1]
```diff
@@ -329,7 +329,7 @@ Transport settings
                                side car connection if needed.
                                Use the format 'PubSubComponent=<PubSubComponent>
                                [;GrpcPort=<GrpcPort>;HttpPort=<HttpPort>[;
-                               Scheme=<'https'|'http'>]][;ApiKey=<ApiKey>]'.
+                               Scheme=<'https'|'http'>][;Host=<Host>]][;ApiKey=<ApiKey>]'.
```
Comment (marcschier) [inline docs/opc-publisher/commandline.md]: Could you also update the CLI OPT help entry in commandline.cs?

**Review by marcschier — CHANGES_REQUESTED** [review]

File: docs/opc-publisher/commandline.md (hunk 1 above)
Comment (Haeavar) [reply docs/opc-publisher/commandline.md]: Done.

**Review by marcschier — APPROVED** [review]

**Pull request merged**